__tests__/ncnewsapp.test.js

```diff
@@ -87,20 +87,117 @@ describe('/api/articles', () => {
+          expect(article).toHaveProperty('votes', expect.any(Number));
+          expect(article).toHaveProperty('article_img_url', expect.any(String));
+          expect(article).toHaveProperty('comment_count', expect.any(String));
+
+        })
       });
   })
+  test('GET:200 the articles should be returned in descending order of created_at ', () => {
+    return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles).toBeSortedBy("created_at", { descending: true });
+      });
+    });
 });
+describe('GET /api/articles/:article_id/comments', () => {
+  test('200: Responds with an array of comments for the given article_id of which each comment should have the specified properties', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body;
+        expect(Array.isArray(comments)).toBe(true);
+        comments.forEach((comment) => {
+          expect(comment).toHaveProperty('comment_id');
+          expect(comment).toHaveProperty('author');
+          expect(comment).toHaveProperty('article_id');
+          expect(comment).toHaveProperty('votes');
+          expect(comment).toHaveProperty('created_at');
+          expect(comment).toHaveProperty('body');
+        });
+      });
+  });
 
+test('200: Comments should be served with the most recent comments first. ', () => {
+  return request(app)
+    .get('/api/articles/3/comments')
+    .expect(200)
+    .then((response) => {
+      expect(response.body.comments).toBeSortedBy("created_at", { descending: true });
+    });
+  });
 
-
+  test('GET:404 sends an appropriate error message when given a valid but non-existent id', () => {
+    return request(app)
+      .get('/api/articles/999/comments')
+      .expect(404)
+      .then((response) => {
+        expect(response.body.message).toBe('article does not exist');
+      });
+  });
+  test('GET:400 sends an appropriate error message when given an invalid id', () => {
+    return request(app)
+      .get('/api/articles/banana/comments')
+      .expect(400)
+      .then((response) => {
+        expect(response.body.message).toBe("Bad request");
+      });
+  });
+});
+  describe('POST /api/articles/:article_id/comments', () => {
+  test('POST:201 inserts a new comment to the db and sends the new comment back to the client', () => {
+    const newComment = {
+      username: 'butter_bridge',
+      body: 'This is amazing'
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        expect(response.body.comment).toEqual(   {
+          comment_id: 19,
+          body: 'This is amazing',
+          article_id: 1,
+          author: 'butter_bridge',
+          votes: 0,
+          created_at: expect.any(String)
+        } );
+      });
```

Comment:
We should add another test, very similar to the first one but with an extra property on the comment, like:  {
      username: 'butter_bridge',
      body: 'This is amazing',
      fruit: 'apple'
    } And we should ignore the extra one and post the comment with a 201 🙂